include/eve/traits/overload/default_behaviors.hpp

```diff
@@ -0,0 +1,157 @@
+  template< template<typename> class Func
+          , typename OptionsValues
+          , typename... Options
+          >
+  struct callable : decorated_with<OptionsValues, Options...>
+  {
+    using base = decorated_with<OptionsValues, Options...>;
+
+    template<typename T>
+    auto operator[](T t) const
+    requires( requires(base const& b) { b[t];} )
+    {
+      auto new_traits = base::operator[](t);
+      return  Func<decltype(new_traits)>{new_traits};
+    }
+
+    template<typename T> void operator[](T t) const =delete;
+
+    template<typename... Args>
+    constexpr auto behavior(auto arch, Args&&... args) const
+    {
+      return Func<OptionsValues>::deferred_call(arch, EVE_FWD(args)...);
+    }
+  };
+
+  //====================================================================================================================
+
+  //====================================================================================================================
+  template< template<typename> class Func
+          , typename OptionsValues
+          , typename... Options
+          >
+  struct elementwise_callable : callable<Func, OptionsValues, conditional_option, Options...>
+  {
+    Func<OptionsValues> const& current() const { return static_cast<Func<OptionsValues>>(*this); }
+
+    template<callable_options O, typename T, typename... Ts>
+    constexpr auto behavior(auto arch, O const& opts, T x0,  Ts const&... xs) const
+    {
+      using func_t                =  Func<OptionsValues>;
+      constexpr bool is_supported = requires{ func_t::deferred_call(arch, opts, x0, xs...); };
+      constexpr bool any_simd     = (simd_value<T> || ... || simd_value<Ts>);
+
+      // Are we dealing with a no-condition call ?
+      if constexpr( option_type_is<condition_key, O, ignore_none_> )
+      {
+        // If a deferred call is present, let's call it
+        if      constexpr( is_supported )  return func_t::deferred_call(arch, opts, x0, xs...);
+        else if constexpr( any_simd )
+        {
+          // if not, try to slice/aggregate from smaller wides or to map the scalar version
+          if constexpr((has_aggregated_abi_v<Ts> || ...)) return aggregate(current(), x0, xs...);
+          else                                            return map(current(), x0, xs...);
+        }
+      }
+      else
+      {
+        // Grab the condition
+        auto const cond     = opts[condition_key];
+
+        // Drop the conditional and process the function call with the universal masker
+        auto const rmv_cond = options{rbr::drop(condition_key, opts)};
+        auto const f        = Func<decltype(rmv_cond)>{rmv_cond};
+
+        // If the conditional call is supported, call it
+        if      constexpr( is_supported ) return detail::mask_op(cond, f, x0, xs...);
+        // if not, call the non-masked version then mask piecewise
+        else if constexpr( any_simd )     return detail::mask_op(cond, detail::eval_2nd, x0, f(x0,xs...));
```

Comment:
Why not just if_else?

Reply:
mask_op is written so all the necessary thing are done properly for the masking.
It is what we use everywhere already.

Reply:
For reference :

```c++
  template<conditional_expr C, typename Op, typename Arg0, typename... Args>
  EVE_FORCEINLINE auto mask_op( C const& c
                              , [[maybe_unused]] Op f
                              , [[maybe_unused]] Arg0 const& a0
                              , [[maybe_unused]] Args const&... as
                              )
  {
    using r_t       = decltype(f(a0,as...));
    auto const cond = c.mask(eve::as<r_t>());

          if constexpr( C::is_complete && !C::is_inverted ) return alternative(c,a0,eve::as<r_t>{});
    else  if constexpr( C::is_complete &&  C::is_inverted ) return f(a0,as...);
    else                                                    return if_else( cond
                                                                          , f(a0,as...)
                                                                          , alternative(c,a0,eve::as<r_t>{})
                                                                          );
  }
  ```